src/datalinker/dependencies/hips.py

```diff
@@ -3,22 +3,134 @@
 import re
 from typing import Annotated
 
-from fastapi import Depends
+from fastapi import Depends, HTTPException
 from httpx import AsyncClient
+from pydantic import HttpUrl
 from safir.dependencies.http_client import http_client_dependency
 from safir.dependencies.logger import logger_dependency
 from structlog.stdlib import BoundLogger
 
-from ..config import config
+from ..config import Config
 from ..constants import HIPS_DATASETS
+from .config import config_dependency
 
 __all__ = [
+    "DatasetHiPSListDependency",
     "HiPSListDependency",
+    "dataset_hips_list_dependency",
     "hips_list_dependency",
 ]
 
 
-class HiPSListDependency:
+class BaseHiPSListDependency:
+    """Base class with common HiPS list functionality."""
+
+    async def _fetch_and_process_properties(
+        self,
+        base_url: HttpUrl,
+        hips_path: str,
+        client: AsyncClient,
+        logger: BoundLogger,
+        config: Config,
+        dataset: str | None = None,
+    ) -> str | None:
+        """Fetch and process a single HiPS properties file.
+
+        Parameters
+        ----------
+        base_url
+            Base URL for the HiPS service.
+        hips_path
+            Path to the specific HiPS dataset.
+        client
+            HTTP client for making requests.
+        logger
+            Logger for error messages.
+        config
+            Configuration object containing auth token.
+        dataset
+            Optional dataset name for logging context.
+
+        Returns
+        -------
+        str | None
+            Processed properties data, or None if fetch failed.
+        """
+        url = str(base_url).rstrip("/") + f"/{hips_path}"
+        r = await client.get(
+            url + "/properties",
+            headers={"Authorization": f"bearer {config.token}"},
+        )
+        if r.status_code != 200:
+            log_msg = "Unable to get HiPS list"
+            log_context = {
+                "url": url,
+                "status": r.status_code,
+                "error": r.reason_phrase,
+            }
+            if dataset:
+                log_msg += " for dataset"
+                log_context["dataset"] = dataset
+
+            logger.warning(log_msg, **log_context)
+            return None
+
+        # Our HiPS properties files don't contain the URL
+        # (hips_service_url), but this is mandatory in the entries in the
+        # HiPS list.  Add it before hips_status.
+        service_url = "{:25}= {}".format("hips_service_url", url)
+        return re.sub(
+            "^hips_status",
+            f"{service_url}\nhips_status",
+            r.text,
+            flags=re.MULTILINE,
+        )
+
+    async def _build_hips_list_from_paths(
```

Comment:
Python unfortunately has no real convention for protected methods that are called by subclasses but not external to the class, just an underscore for private and no underscore for public. I usually leave off the underscore for methods that are expected to be called by subclasses, since they're part of the interface of the class.